src/ITensorNetworks/MPSTensor/GeneralMPSTensor.jl

```diff
@@ -0,0 +1,66 @@
+using ChainRulesCore
+using ZygoteRules: @adjoint
+
+struct GeneralMPSTensor <: AbstractNetwork
+  mps::MPS
+end
+
+function Base.:+(A::GeneralMPSTensor, B::GeneralMPSTensor; kwargs...)
+  if length(A.mps) == 1 && order(A.mps[1]) == 0
+    return GeneralMPSTensor(MPS([ITensor(A[] + B[])]))
+  else
+    return GeneralMPSTensor(+(A.mps, B.mps; kwargs...))
+  end
+end
+
+function Base.getindex(t::GeneralMPSTensor)
+  @assert length(t.mps) == 1
+  @assert order(t.mps[1]) == 0
+  return t.mps[1][]
+end
+
+function ChainRulesCore.rrule(::typeof(getindex), x::GeneralMPSTensor)
+  y = x[]
+  function getindex_pullback(ȳ)
+    x̄ = GeneralMPSTensor(MPS([ITensor(ȳ)]))
+    return (NoTangent(), x̄)
+  end
+  return y, getindex_pullback
+end
+
+"""
+initialization methods
+"""
+Base.isless(a::ITensors.Index, b::ITensors.Index) = a.id < b.id
+
+function GeneralMPSTensor(tensor::ITensor; cutoff, maxdim)
+  mps_out = if size(tensor) == ()
+    MPS([tensor])
+  else
+    MPS(tensor, sort(inds(tensor)); cutoff=cutoff, maxdim=maxdim)
```

Comment:
Is there a good reason for sorting here? I think using `inds(tensor)` is good because it doesn't require as much swapping to perform the SVDs.

Reply:
For two tensors t1 and t2 having the same indices, inds(t1) and inds(t2) might output indices with different order. This will have trouble when we do `MPS(t1)+MPS(t2)`, since the indices are not consistent.

Reply:
I see. But this seems to only solve that problem in a very particular case (where you happened to form the MPS from an ITensor). Won't there be other cases where the MPS are formed in different ways and therefore may be unordered? Is it better to just fix MPS addition for the case when the MPS are out of order? It would be relatively easy, we have a general MPS swap functionality (there is a function called `movesites`, for example).

Reply:
In general, I think given an itensor with specific inds, we always want the MPS generated to have a fixed index order. The addition is one example where this can be important. Another example is if we want to do a simple inner product, `mps(t1) * mps(t2)`, it doesn't make sense to do a lot of swaps, which will hurt the performance. And for this function that transforms an ITensor to MPSTensor, I don't think it will need swaps, since what it does is just to transpose the input tensor and then do SVDs.

For the case of MPSTensor times a MPSTensor, here I also use this sort, but it's just an implementation to get the code working, we will definitely need to change the implementations there.

Reply:
Ok, seems reasonable for now. I just wanted to understand what the thought process was and see if there is a more general solution to this problem (since usually relying on a specific index ordering of ITensors is bad practice, but MPS are exceptional because of how much work it is to permute them). Probably it is helpful to define `MPS + MPS` and `MPS * MPS` for MPS with permuted site indices from each other, but agreed it is best to avoid that if possible.